libs/wingc/src/ast.rs

```diff
@@ -188,11 +188,15 @@ pub struct FunctionDefinition {
 	/// List of names of function parameters and whether they are reassignable (`var`) or not.
 	pub parameters: Vec<(Symbol, bool)>, // TODO: move into FunctionSignature and make optional
 	/// The function implementation (body).
-	pub statements: Scope,
+	pub statements: Option<Scope>,
```

Comment:
From a modeling perspective, since 
 `statements` and `external_implementation` are be mutually exclusive, I'd model them as an enumeration:

```rust
body: FunctionBody;


// and then
enum FunctionBody {
  External(...),
  Statements(...)
}
```